clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -3824,6 +4506,10 @@ class SyclKernelIntHeaderCreator : public SyclKernelFieldHandler {
     return handleScalarType(FD, FieldTy);
   }
 
+  bool handleUnionType(ParmVarDecl *PD, QualType ParamTy) final {
+    return handleScalarType(PD, ParamTy);
```

Comment:
Please replace with //TODO and unsupportedFreeFunctionParamType()

Reply:
Done.